### PR TITLE
LSF norm and ranges

### PR DESCRIPTION
**Pull request by jnburchett**
Fixed normalization of LSF in several places.
Input wavelengths above and below defined LSF wavelengths now allowed

### CODE REVIEW

File: linetools/spectra/lsf.py [hunk 1]
```diff
@@ -230,17 +231,36 @@ def interpolate_to_wv0(self,wv0):
...
                 aux_val += [row[i]]
-            f = interp1d(col_waves,aux_val,bounds_error=True,kind='linear') #we do not want to extrapolate
-            lsf_vals += [f(wv0.value)]
+            #we don't want to extrapolate wildly, but allow small extrapolations for wv0
+            if (wv0.value >= col_waves[0]) & (wv0.value <= col_waves[-1]):
+                f = interp1d(col_waves,aux_val,bounds_error=True,kind='linear') #no need to extrapolate
+                lsf_vals += [f(wv0.value)]
+
+            elif (wv0.value < col_waves[0]) & \
+                    ((col_waves[0]-wv0.value) < np.abs(col_waves[1]-col_waves[0])):
```
Comment (ntejos) [inline linetools/spectra/lsf.py]: Can you limit the extrapolation to `np.abs(col_waves[1]-col_waves[0])) / 2` , i.e. half way only.

File: linetools/spectra/lsf.py [hunk 2]
```diff
@@ -230,17 +231,36 @@ def interpolate_to_wv0(self,wv0):
...
+                f = interp1d(col_waves, aux_val, bounds_error=False,
+                             fill_value=aux_val[0], kind='linear')  #assign shortest wv LSF definition
+                lsf_vals += [f(wv0.value)]
+
+            elif (wv0.value > col_waves[-1]) & ((wv0.value - col_waves[-1]) < np.abs(col_waves[-1] - col_waves[-2])):
```
Comment (ntejos) [inline linetools/spectra/lsf.py]: Ditto here.

File: linetools/spectra/lsf.py [hunk 3]
```diff
@@ -230,17 +231,36 @@ def interpolate_to_wv0(self,wv0):
...
+                f = interp1d(col_waves, aux_val, bounds_error=False,
+                             fill_value=aux_val[-1], kind='linear') #assign longest wv LSF definition
+                lsf_vals += [f(wv0.value)]
+
+            else:
+                raise ValueError("wv0 too far outside range of defined LSFs.  Perhaps you've chosen the wrong grating?")
         lsf_vals = np.array(lsf_vals)
-        #normalize
-        lsf_vals /= np.max(lsf_vals)
+
```
Comment (ntejos) [inline linetools/spectra/lsf.py]: Could you add tests to check the extrapolation is behaving as expected?

File: linetools/spectra/lsf.py (hunk 1 above)
Comment (jnburchett) [inline linetools/spectra/lsf.py]: Done.

File: linetools/spectra/lsf.py (hunk 2 above)
Comment (jnburchett) [inline linetools/spectra/lsf.py]: Done.

File: linetools/spectra/lsf.py (hunk 3 above)
Comment (jnburchett) [inline linetools/spectra/lsf.py]: Making sure that joebvp is behaving as expected with these new changes first.

**Comment by ntejos** [comment]
Also, there seems to be a conflict with master in current branch; please resolve.

**Comment by jnburchett** [comment]
Any guidance on how to track this down?

File: linetools/spectra/lsf.py (hunk 2 above)
Comment (jnburchett) [inline linetools/spectra/lsf.py]: Uh-oh, just testing on CASBaH data that extend to 1800 Angstroms.  The LSF definition file only goes to 1750.  I'm extending the range on long-wavelength end.  In general, the NUV will take over beyond 1800 \AA, but we have lines right out to 1800 covered by G160M.  I'll leave the shorter extrapolation range on the short-wavelength end (~1150 \AA).

Comment (ntejos) [inline linetools/spectra/lsf.py]: Sounds reasonable. Could you add a warning message saying that we are extrapolating the COS LSF by XX A from their closest tabulated one?

Comment (ntejos) [inline linetools/spectra/lsf.py]: Same warning message for the blue end please.

File: linetools/spectra/lsf.py (hunk 3 above)
Comment (jnburchett) [inline linetools/spectra/lsf.py]: Added.  However, I still get odd behavior in joebvp due to the negative LSF values.  Can we add these fixes once you review this PR?

Comment (ntejos) [inline linetools/spectra/lsf.py]: Sure, will add the fix as soon as we merge this PR.

**Pull request merged**